File: DS-Algo-with-Java/Lecture-8/AddBinary.java
import java.util.Scanner;

public class AddBinary {
    public static void main(String[] args) {
        String binary1, binary2;
        try (Scanner sc = new Scanner(System.in)) {
            System.out.print("Enter 1st binary number: ");
            binary1 = sc.next();
            System.out.print("Enter 2nd binary number: ");
            binary2 = sc.next();

            int num1 = Integer.parseInt(binary1, 2);
            int num2 = Integer.parseInt(binary2, 2);
            int sum = num1 + num2;

            String result = Integer.toBinaryString(sum);
            System.out.print(binary1 + " + " + binary2 + " = " + result);
        }
    }
}

File: DS-Algo-with-Python/Lecture-8/AddBinary.py
binary1 = input("Enter 1st binary number: ")
binary2 = input("Enter 2nd binary number: ")

num1 = int(binary1, 2)
num2 = int(binary2, 2)
addition = num1 + num2

result = bin(addition).replace("0b", "")
print(f"{binary1} + {binary2} = {result}")
